edtf/parser/grammar.py

```diff
@@ -340,14 +343,22 @@ def f(toks):
 )
 
 
-def parse_edtf(str, parseAll=True, fail_silently=False):
+def parse_edtf(input_string, parseAll=True, fail_silently=False, debug=None):
+    if debug is None:
+        debug = DEBUG_PYPARSING
     try:
-        if not str:
+        if not input_string:
             raise ParseException("You must supply some input text")
```

Comment:
I think maybe this should be moved up out of the `try:` block, because otherwise you don't see the "You must supply..." message.